### PR TITLE
output: allow arbitrary keys in metadata

### PR DESCRIPTION
**Pull request by regit**
Metadata keyword in signatures can have any key defined so we
should allow them.

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- update schema.json

### CODE REVIEW

**Comment by victorjulien** [comment]
Interesting issue. I feel this should be restrictive in our CI, but I see that outside of our CI it should not be.

**Comment by regit** [comment]
Just for information: I've added a commit to the MR as it is on the same subject.

**Comment by regit** [comment]
> Interesting issue. I feel this should be restrictive in our CI, but I see that outside of our CI it should not be.

Should it be the suricata-verify test that are restrictive and not the schema check ?

**Comment by victorjulien** [comment]
> > Interesting issue. I feel this should be restrictive in our CI, but I see that outside of our CI it should not be.
> 
> Should it be the suricata-verify test that are restrictive and not the schema check ?

New SV tests or SV tests exposing new EVE fields trigger schema error if fields are not part of the schema. This has been a major driver of schema updates.

**Comment by catenacyber** [comment]
I stumbled on the metadata schema.json problem with https://github.com/OISF/suricata/pull/8320 ;-)

**Comment by catenacyber** [comment]
Eric, do you have S-V tests for the `port` field ?

File: etc/schema.json [hunk 1]
```diff
@@ -244,26 +244,23 @@
...
-                        },
                         "updated_at": {
                             "type": "array",
                             "items": {
                                 "type": "string"
                             }
                         }
                     },
-                    "additionalProperties": false
+                    "additionalProperties": true
```
Comment (jasonish) [inline etc/schema.json:254]: This is already done in master.

File: etc/schema.json [hunk 2]
```diff
@@ -244,26 +244,23 @@
                                 "type": "string"
                             }
                         },
-                        "tag": {
-                            "type": "array",
-                            "items": {
-                                "type": "string"
-                            }
-                        },
```
Comment (jasonish) [inline etc/schema.json:252]: Not sure why this is needed.

**Comment by jasonish** [comment]
Replace by https://github.com/OISF/suricata/pull/8824.

**Pull request closed without merging**